### PR TITLE
[habana] Misc fixes for vision nets

### PR DESCRIPTION
**Pull request by bertmaher**
* Introduced LRN, fixed #144
* Added Concat node to the list of supported quantized nodes
* Added RescaleQuantized node, replaced quantized MatMul with GEMM

### CODE REVIEW

File: lib/Backends/Habana/Habana.cpp [hunk 1]
```diff
@@ -552,10 +552,10 @@ makeSynPoolParams(llvm::ArrayRef<unsigned_t> kernel,
   params->sW = stride[0];
   params->sH = stride[1];
   // Padding
-  params->pWbegin = pad[0];
-  params->pWend = pad[0];
-  params->pHbegin = pad[1];
-  params->pHend = pad[1];
+  params->pHbegin = pad[0];
```
Comment (SplitInfinity) [inline lib/Backends/Habana/Habana.cpp]: Lol, whoops.

File: lib/Backends/Habana/Habana.cpp [hunk 2]
```diff
@@ -1011,6 +1027,18 @@ HabanaBackend::compile(Function *F, const BackendOptions &opts) const {
...
       break;
     }
+    case Kinded::Kind::LocalResponseNormalizationNodeKind: {
+      auto *NI = llvm::cast<LocalResponseNormalizationNode>(&I);
+      std::unique_ptr<ns_LrnKernel::Params> params = makeLrnParams(
+          NI->getAlpha(), NI->getBeta(), NI->getK(), NI->getHalfWindowSize());
+
+      chk(synCreateGenericNode(&tensors[NI->getInput()].get(),
+                               &tensors[NI].get(), 1, 1, (void *)params.get(),
+                               "lrn_f32", NI->getName().str().c_str(), nullptr,
```
Comment (SplitInfinity) [inline lib/Backends/Habana/Habana.cpp]: Is the datatype intentionally hardcoded? Does this operator not exist for other datatypes? It makes sense that it wouldn't, but I just want to confirm.

**Comment by SplitInfinity** [comment]
Also, it's interesting that we need LRN because I haven't heard of it being used for months now.

File: lib/Backends/Habana/Habana.cpp (hunk 2 above)
Comment (bertmaher) [reply lib/Backends/Habana/Habana.cpp]: Good eye - LRN does indeed only exist for f32.  We could assert that here, although it'll already be caught by isOpSupported.

**Comment by bertmaher** [comment]
It's used in inception_v1 and bvlc_alexnet.  Unless it usually gets optimized away somehow?

**Comment by SplitInfinity** [comment]
Ah, those two are 4-8 years old at this point. I thought some recent models use them.

**Comment by sergeigofman** [comment]
LRN indeed aren't used in modern topologies. I introduced it more like an exe4cise and to enable inception_v1. Besides, Habana supports it so why not? Indeed it is only available for fp32.

**Pull request merged**